dags/backfill-bqetl.py

```diff
@@ -0,0 +1,104 @@
+import ast
+import dataclasses
+import datetime
+
+from typing import Optional, List
+from airflow.decorators import dag
+from airflow.operators.python import PythonOperator
+from utils.tags import Tag
+
+
+@dataclasses.dataclass
+class BackfillParams:
+    destination_table: str
+    dataset_id: str
+    start_date: str
+    end_date: str
+    project_id: str
+    excludes: List[str]
+    dry_run: bool
+    parallelism: int
+    no_partition: bool
+
+    def validate_date_range(self) -> None:
+        start_date = datetime.datetime.fromisoformat(self.start_date)
+        end_date = datetime.datetime.fromisoformat(self.end_date)
+        if start_date > end_date:
+            raise ValueError(f"`start_date`={self.start_date} is greater than `end_date`={self.end_date}")
+
+
+def __parse_string_params(string_params: str) -> Optional[BackfillParams]:
+    """
+    dag_run.conf is string representation of a Python dictionary in Airflow 2.1
+    ast.literal_eval() is used to convert from string to dictionary as a workaround
+    this workaround will no longer be required in Airflow >=2.2, see link below for future implementation
+    https://airflow.apache.org/docs/apache-airflow/stable/concepts/params.html
+    """
+    params_parsed = ast.literal_eval(string_params)
+    return BackfillParams(**params_parsed)
+
+
+def param_validation(params: str) -> bool:
+    backfill_params = __parse_string_params(params)
+    backfill_params.validate_date_range()
+    return True
```

Comment:
This is no longer required; we are using Airflow 2.3 which supports the `Param` model, see current backfill DAG implementation.